src/include/detail/linalg/tdb_io.h

```diff
@@ -376,63 +415,40 @@ auto sizes_to_indices(const std::vector<T>& sizes) {
  */
 template <class T>
 auto read_bin_local(const std::string& bin_file, size_t subset = 0) {
-  if (!std::filesystem::exists(bin_file)) {
-    throw std::runtime_error("file " + bin_file + " does not exist");
-  }
-  auto file_size = std::filesystem::file_size(bin_file);
-
-  auto fd = open(bin_file.c_str(), O_RDONLY);
-  if (fd == -1) {
-    throw std::runtime_error("could not open " + bin_file);
-  }
+  std::uint32_t dimension = 0u;
 
-  uint32_t dimension{0};
-  auto num_read = read(fd, &dimension, 4);
-  lseek(fd, 0, SEEK_SET);
+  const auto file_size = std::filesystem::file_size(bin_file);
+  auto file = std::ifstream{bin_file, std::ios::binary};
```

Comment:
(Unsure) Do we need to add a `if (!file.is_open()) {` check? I see they do that here: https://en.cppreference.com/w/cpp/io/basic_ifstream

Reply:
I think that we should add check for each read. The read would inevitably fail if the file failed to open

Reply:
Sorry, don't follow. You're saying the caller should check something after calling `read_bin_local()`?

Reply:
Ah sorry my bad. I assumed that the `std::iostream` constructor would throw an exception (and it actually does) so it would never get to checking `file.open` if it failed to open that file. But it would not hurt to add this check so I'll do it.